bolt-sidecar/benches/score_cache.rs

```diff
@@ -0,0 +1,81 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use std::collections::HashMap;
+
+use bolt_sidecar::common::score_cache::ScoreCache;
+
+const GET_SCORE: isize = 1;
+const INSERT_SCORE: isize = 2;
+const UPDATE_SCORE: isize = 3;
+
+fn bench_scorecache_vs_hashmap(c: &mut Criterion) {
+    let mut group = c.benchmark_group("ScoreCache vs HashMap");
+
+    let mut score_cache: ScoreCache<GET_SCORE, INSERT_SCORE, UPDATE_SCORE, usize, usize> =
+        ScoreCache::with_max_len(10_000);
+    let mut hash_map: HashMap<usize, usize> = HashMap::new();
```

Comment:
I think each benchmark iteration should be isolated. Now it's sharing `score_cache` and `hash_map` which might skew the results